common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleClient.java

```diff
@@ -42,5 +42,13 @@ public abstract void fetchBlocks(
       String execId,
       String[] blockIds,
       BlockFetchingListener listener,
-      File[] shuffleFiles);
+      boolean toDisk,
+      TmpFileCreater tmpFileCreater);
+
+  /**
+   * An interface to provide approach to create tmp file.
+   */
+  public interface TmpFileCreater {
```

Comment:
Actually we may not need this, just let `ExternalShuffleClient.fetchBlocks` accept a java lambda.